### ISSUE TITLE
`findFirst` with nested selection into explicit many-to-many table not working

### ISSUE THREAD

**Issue by electrovir** [morintd/prismock#993]
I have a query like this:

```typescript
const result = await prismaClient.user.findFirst({
    select: {
        id: true,
        name: true,
        manyToManyTable: {
            select: {
                otherRelation: {
                    select: {
                        id: true,
                        name: true,
                    }
                }
            }
        }
    }
})
```

Where my Prisma schema looks like this:

```prisma
model User {
    id              Int
    name            String
    manyToManyTable ManyToManyTable[]
}

model ManyToManyTable {
    userId          Int
    otherRelationId Int
    
    otherRelation OtherRelation @relation(fields: [otherRelationId], references: [id])
    user          User          @relation(fields: [userId], references: [id])
}

model OtherRelation {
    id              Int
    name            String
    manyToManyTable ManyToManyTable[]
}
```

The mentioned query always returns an empty array for `result.manyToManyTable`, even when I've verified that the data exists.

Is this a known limitation? I didn't see anything obviously relevant in the README. If you need a repro example repo let me know and I'll whip one up.

**Comment by morintd** [comment]
Hello @electrovir , this is not a known limitation, but I didn't really test with many-to-many relationships. I'll have a look to do so, and try to find a fix for this use-case.

**Comment by electrovir** [comment]
Fyi, if you need/want any help with this, I'm willing to dig in.

**Comment by electrovir** [comment]
Possibly relevant: https://github.com/morintd/prismock/issues/1016

Maybe the relations weren't being created properly in the first place? I'll have to try the new release.

**Comment by morintd** [comment]
Hello @electrovir , from the test in https://github.com/morintd/prismock/pull/1065, it seems like nested selection into explicit many-to-many table are working fine. Are you able to add a reproducible example/test?

From my investigation on many-to-many relation, I noticed we triggered a duplicate include, which could end up with a wrong return data. So, thanks for helping discovering that!

**Comment by electrovir** [comment]
I'll spin up an repro repo in the next few days.

**Comment by electrovir** [comment]
Good news! While trying to create a minimal reproduction of the issue, I've discovered that it's been fixed in 1.33.1 🥳 (this issue was experience in 1.30.2). Thank you!
